servers/zms/src/main/java/com/yahoo/athenz/zms/store/file/FileConnection.java

```diff
@@ -405,6 +405,34 @@ public boolean insertDomainTemplate(String domainName, String templateName, Stri
         return true;
     }
 
+    @Override
+    public boolean updateDomainTemplate(String domainName, String templateName, TemplateMetaData templateMetaData) {
+        DomainStruct domainStruct = getDomainStruct(domainName);
+        ArrayList<TemplateMetaData> templateMetalist = new ArrayList<>();
+        if (domainStruct == null) {
+            throw ZMSUtils.error(ResourceException.NOT_FOUND, "domain not found", "updateDomainTemplate");
+        }
+        if (domainStruct.getTemplates() == null) {
+            domainStruct.setTemplates(new ArrayList<>());
+        }
+        ArrayList<String> templates = domainStruct.getTemplates();
+        if (!templates.contains(templateName)) {
+            templates.add(templateName);
+        }
+        TemplateMetaData templateMeta = new TemplateMetaData();
+        templateMeta.setTemplateName(templateName);
+        templateMeta.setKeywordsToReplace(templateMetaData.getKeywordsToReplace());
+        templateMeta.setAutoUpdate(templateMetaData.getAutoUpdate());
+        templateMeta.setDescription(templateMetaData.getDescription());
+        templateMeta.setTimestamp(templateMetaData.getTimestamp());
+        templateMeta.setCurrentVersion(templateMetaData.getLatestVersion());
+        templateMetalist.add(templateMeta);
+        domainStruct.setTemplateMeta(templateMetalist);
```

Comment:
we're creating the template meta list every time this api is called and replacing the struct in domain struct. which means it will only have a single template and will be overwritten every time. The implementation needs to be:

if (domainStruct.getTemplateMeta() == null) {
             domainStruct.setTemplateMeta(new ArrayList<>());
 }
ArrayList<TemplateMetaData> templateMetalist = domainStruct.getTemplateMeta();

// then iterate through the list and see if we have an object
// with that template. If yes, update the version only,
// if not then create a new object and add it to the list